File: server/model/src/main/kotlin/hu/bsstudio/bssweb/video/model/Video.kt
package hu.bsstudio.bssweb.video.model

import java.time.LocalDate
import java.util.UUID

data class Video(
    val id: UUID,
    val title: String,
    val urls: List<String>,
    val description: String,
    val shootingDateStart: LocalDate,
    val shootingDateEnd: LocalDate,
    val visible: Boolean,
)

File: integration/src/types/video.ts
import type { UUID } from './common.js';

export interface CreateVideo {
  title: string;
}

export interface UpdateVideo extends CreateVideo {
  urls: string[];
  description: string;
  shootingDateStart: string; // ISO date string
  shootingDateEnd: string; // ISO date string
  visible: boolean;
  labels: string[];
}

export interface Video extends UpdateVideo {
  id: UUID;
}

export interface VideoCrew {
  videoId: UUID;
  position: string;
  member: {
    id: UUID;
    name: string;
    nickname: string;
  };
}

export interface DetailedVideo extends Video {
  crew: VideoCrew[];
}
